plugins/jobs/server/job_rest.py

```diff
@@ -42,30 +46,41 @@ def __init__(self):
                'not passed or empty, will use the currently logged in user. If '
                'set to "None", will list all jobs that do not have an owning '
                'user.', required=False)
+        .param('typesArray', 'Filter for type', required=False)
+        .param('statusesArray', 'Filter for type', required=False)
         .pagingParams(defaultSort='created', defaultSortDir=SortDir.DESCENDING)
     )
-    def listJobs(self, userId, limit, offset, sort, params):
+    def listJobs(self, userId, typesArray, statusesArray, limit, offset, sort, params):
         currentUser = self.getCurrentUser()
         if not userId:
             user = currentUser
         elif userId.lower() == 'none':
             user = None
         else:
-            user = self.model('user').load(userId, user=currentUser, level=AccessType.READ)
+            user = self.model('user').load(
+                userId, user=currentUser, level=AccessType.READ)
 
+        types = json.loads(typesArray) if typesArray is not None else None
+        statuses = json.loads(statusesArray) if statusesArray is not None else None
         return list(self.model('job', 'jobs').list(
-            user=user, offset=offset, limit=limit, sort=sort, currentUser=currentUser))
+            user=user, offset=offset, limit=limit, types=types,
+            statuses=statuses, sort=sort, currentUser=currentUser))
 
     @access.admin
     @filtermodel(model='job', plugin='jobs')
     @autoDescribeRoute(
         Description('List all jobs.')
+        .param('typesArray', 'Filter for type', required=False)
+        .param('statusesArray', 'Filter for type', required=False)
         .pagingParams(defaultSort='created', defaultSortDir=SortDir.DESCENDING)
     )
-    def listAllJobs(self, limit, offset, sort, params):
+    def listAllJobs(self, typesArray, statusesArray, limit, offset, sort, params):
         currentUser = self.getCurrentUser()
-        return list(self.model('job', 'jobs').listAll(
-            offset=offset, limit=limit, sort=sort, currentUser=currentUser))
+        types = json.loads(typesArray) if typesArray is not None else None
+        statuses = json.loads(statusesArray) if statusesArray is not None else None
```

Comment:
We could use the `jsonParam` instead of `param` for these two new parameters since they are JSON.